snaxc/dialects/phs.py

```diff
@@ -0,0 +1,333 @@
+        super().__init__(
+            properties={
+                "sym_name": StringAttr(name),
+            },
+            operands=(lhs, rhs, switch),
+            regions=(default_region, case_regions),
+            attributes=attr_dict,
+            result_types=(result_types,),
+        )
+
+    @property
+    def data_operands(self) -> Sequence[SSAValue]:
+        """
+        Returns all operands except the switch operands
+        """
+        return self._operands[:-1]
+
+    @staticmethod
+    def from_operations(
+        name: str,
+        lhs: Operation | SSAValue,
+        rhs: Operation | SSAValue,
+        switch: Operation | SSAValue,
+        operations: Sequence[type[FloatingPointLikeBinaryOperation]],
+        result_types: Sequence[Attribute] = [],
+    ) -> "ChooseOpOp":
+        """
+        Utility constructor to construct a ChooseOpOp from a set of given operations
+        """
+        # Default operation
+        case_regions: list[Region] = []
+        if len(operations) < 1:
+            for operation in operations[1:]:
+                case_regions.append(
+                    Region(
+                        Block(
+                            [
+                                result := operation(lhs, rhs),
+                                YieldOp(result),
+                            ]
+                        )
+                    )
+                )
+        default_region = Region(
+            Block(
+                [
+                    result := operations[0](lhs, rhs),
+                    YieldOp(result),
+                ]
+            )
+        )
+        return ChooseOpOp(
+            name=name,
+            lhs=lhs,
+            rhs=rhs,
+            switch=switch,
+            default_region=default_region,
+            case_regions=case_regions,
+            result_types=result_types,
+        )
+
+    def insert_operations(self, operations: Sequence[FloatingPointLikeBinaryOperation]):
+        """
+        Add an operation to the list of choices if it is not present yet
+        """
+        # FIXME, what if operation order is swapped? i.e. rhs on lhs side and vice versa?
+        for operation in operations:
+            if operation.name not in [op.name for op in self.operations()]:
+                self.add_region(Region(Block([op := type(operation)(self.lhs, self.rhs), YieldOp(op)])))
+
+    def operations(self) -> Iterator[FloatingPointLikeBinaryOperation]:
+        """
+        Get an iterator over the list of existing choices of operations
+        """
+        for region in self.regions:
+            # Only yield the first operation in the region
+            operation = region.ops.first
+            if isinstance(operation, FloatingPointLikeBinaryOperation):
+                yield operation
+
+
+@irdl_op_definition
+class ChooseInputOp(IRDLOperation):
+    """
+    Operation to select between multiple inputs.
```

Comment:
```suggestion
    Operation to select between two inputs.
```

How would you choose between three inputs?